test/libsolidity/SemanticTest.cpp

```diff
@@ -106,16 +108,26 @@ TestCase::TestResult SemanticTest::run(ostream& _stream, string const& _linePref
 {
 	TestResult result = TestResult::Success;
 	bool compileViaYul = m_runWithYul || m_enforceViaYul;
+	bool compileToEwasm = m_runWithEwasm || m_enforceCompileToEwasm;
```

Comment:
Calling these variables `compileViaYul`/`compileToEwasm` is incredibly confusing. There are too many things called `compileViaYul` in this file that mean slightly different things and it's really hard to understand the code because of that.

They're much closer in meaning to `m_runWithYul`/`m_runWithEwasm` so I'd use that as a base. And I'd reverse the meaning to be able to use a less ambiguous verb. So how about:
```suggestion
	bool skipYulRun = !m_runWithYul && !m_enforceViaYul;
	bool skipWasmRun = !m_runWithEwasm && !m_enforceCompileToEwasm;
```

Or, alternatively, I'd just remove the variables and use them directly in conditions below. I think it would reduce the number of names one has to keep track of and thus improve readability.